include/swift/AST/Types.h

```diff
@@ -5364,6 +5366,33 @@ inline TypeBase *TypeBase::getDesugaredType() {
   return cast<SugarType>(this)->getSinglyDesugaredType()->getDesugaredType();
 }
 
+inline bool TypeBase::hasSimpleTypeRepr() const {
```

Comment:
I wouldn't use TypeRepr in the name here since this has nothing to do with `TypeRepr`s.

I don't have any great ideas on naming this, though.

Reply:
I can't think of a better name either. FWIW – If you look at `TypeRepr::isSimple()`, there is very similar logic. The problem is essentially the same and that is why I named the method as such.

If I add cross-referenceing comments to *both* the `TypeRepr` and `TypeBase`, would you be okay with the name?

Reply:
Sorry I wasn't clear here. This was just a suggestion that the name might be a bit misleading and that a comment on what the method does would be helpful.

I don't have a better suggestion for the name, but I don't think that should hold up merging the change.